openedx/core/djangoapps/zendesk_proxy/utils.py

```diff
@@ -73,7 +88,71 @@ def _std_error_message(details, payload):
                     payload
                 )
             )
+        if additional_info:
+            ticket = json.loads(response.text)['ticket']
+            return post_additional_info_as_comment(ticket['id'], additional_info)
+
         return response.status_code
     except Exception:  # pylint: disable=broad-except
         log.exception(_std_error_message('Internal server error', payload))
         return status.HTTP_500_INTERNAL_SERVER_ERROR
+
+
+def get_zendesk_group_by_name(name):
+    """
+    Calls the Zendesk list-groups api
+
+    Returns the group Id matching the name.
+    """
+    url = urljoin(settings.ZENDESK_URL, '/api/v2/groups.json')
+
+    try:
+        response = requests.get(url, headers=_get_request_headers())
+
+        groups = json.loads(response.text)['groups']
+        for group in groups:
+            if group['name'] == name:
+                return group['id']
+    except Exception as exception:  # pylint: disable=broad-except
```

Comment:
I think we can narrow this exception to either a request error or a json error and provide different messages for each.